src/averageAges.js

```diff
@@ -55,10 +75,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filteredPeople = people.filter(person =>
+    onlyWithSon
+      ? person.sex === SEX_MALE && (person.mother)
+      : person.mother
+  );
+
+  const ageDifferences = filteredPeople
+    .map(child => {
+      const mother = people.find(person => person.name === child.mother);
+
+      return (mother) ? child.born - mother.born : 0;
+    })
+    .filter(difference => difference > 0);
+
+  return ageDifferences
+    .reduce((sum, age) => sum + age, INITIAL_VALUE) / ageDifferences.length;
```

Comment:
Now everything is written correctly, but you make a pass through the array three times, and here you can get by with one (this affects the performance a lot).
To do this, transfer all the logic to the middle of the reduce (search for the mother and calculation of the difference).